### ISSUE TITLE
2.7.10 introduces DNS issues

### ISSUE THREAD

**Issue by ltguillaume** [Windscribe/Desktop-App#104]
1. Updating to 2.7.10 from within the app (coming from 2.7.9) failed: Windscribe exits in order to run the setup executable, but it's never triggered.
2. After manually running the setup executable, the connection always fails (see log below). I'm using ROBERT as DNS. Changing "App Internal DNS" does not help. Switching back to 2.7.9 immediately resolves the issue.
```
[connection]	 Connection settings: {isAutomatic:false; protocol:"WireGuard"; port:443}
[connection]	 "Location nodes: node1 = {ip1 (...)}; "
[connection]	 Default adapter and gateway: "adapter name = Intel(R) Gigabit Network Connection; adapter IP = 192.168.?.?; gateway IP = 192.168.?.?; remote IP = ; dns = (192.168.?.?); ifIndex = 19"
[connection]	 Connecting to IP: "(...)"  protocol: "WireGuard"  port: 443
[connection]	 Requesting WireGuard config for hostname = "(...).whiskergalaxy.com"
[basic]	 update session status (changed since last call)
[basic]	 Received Check Update Answer
[basic]	 No available update
[server_api]	 API request PortMap successfully executed
[server_api]	 API request ServerCredentials "ikev2" successfully executed
[server_api]	 API request ServerConfigs successfully executed
[server_api]	 Notifications request successfully executed
[server_api]	 WgConfigs connect request successfully executed
[basic]	 Whitelist connecting ip: "(...)"
[basic]	 Whitelist DNS-server ip: QList("10.255.255.4")
[firewall_controller]	 firewall enabled with ips count: 181
[server_api]	 API request ServerCredentials "openvpn" successfully executed
[connection]	 Starting "WireguardService"
[server_api]	 API request ServerLocations successfully executed, revision changed = 11708 , revision_hash = "(...)"
[basic]	 Servers locations changed
[connection]	 Opened and mapped the wireguard service log file
[wireguard]	 "140923 14:21:49:095" "[TUN] [WindscribeWireguard] Starting WireGuard/0.5.3 (Windows 10.0.?????; amd64)"
[wireguard]	 "140923 14:21:49:095" "[TUN] [WindscribeWireguard] Watching network interfaces"
[wireguard]	 "140923 14:21:49:097" "[TUN] [WindscribeWireguard] Resolving DNS names"
[wireguard]	 "140923 14:21:49:097" "[TUN] [WindscribeWireguard] Creating network adapter"
[wireguard]	 "140923 14:21:49:139" "[TUN] [WindscribeWireguard] Using existing driver 0.10"
[wireguard]	 "140923 14:21:49:145" "[TUN] [WindscribeWireguard] Creating adapter"
[wireguard]	 "140923 14:21:49:236" "[TUN] [WindscribeWireguard] Using WireGuardNT/0.10"
[wireguard]	 "140923 14:21:49:236" "[TUN] [WindscribeWireguard] Enabling firewall rules"
[wireguard]	 "140923 14:21:49:222" "[TUN] [WindscribeWireguard] Interface created"
[wireguard]	 "140923 14:21:49:238" "[TUN] [WindscribeWireguard] Dropping privileges"
[wireguard]	 "140923 14:21:49:238" "[TUN] [WindscribeWireguard] Setting interface configuration"
[wireguard]	 "140923 14:21:49:238" "[TUN] [WindscribeWireguard] Peer 1 created"
[wireguard]	 "140923 14:21:49:239" "[TUN] [WindscribeWireguard] Monitoring MTU of default v4 routes"
[wireguard]	 "140923 14:21:49:239" "[TUN] [WindscribeWireguard] Interface up"
[wireguard]	 "140923 14:21:49:239" "[TUN] [WindscribeWireguard] Setting device v4 addresses"
[wireguard]	 "140923 14:21:49:287" "[TUN] [WindscribeWireguard] Startup complete"
[wireguard]	 "140923 14:21:49:457" "[TUN] [WindscribeWireguard] Sending handshake initiation to peer 1 (...)"
[wireguard]	 "140923 14:21:49:696" "[TUN] [WindscribeWireguard] Receiving handshake response from peer 1 (...)"
[wireguard]	 "140923 14:21:49:696" "[TUN] [WindscribeWireguard] Keypair 1 created for peer 1"
[connection]	 ConnectionManager::onConnectionConnected(), state_ = 1
[connection]	 VPN adapter and gateway: "adapter name = WireGuard Tunnel; adapter IP = 100.86.191.?; gateway IP = 0.0.0.0; remote IP = ; dns = (10.255.255.4); ifIndex = 24"
[firewall_controller]	 firewall enabled with ips count: 1
[basic]	 IPv6 disabled
[packet_size]	 Packet size mode auto - using default MTU (Engine)
[basic]	 Close all active TCP connections (keepLocalSockets =  true )
[default]	 MeasurementCpuUsage detect CPU usage in connected state
[connection]	 TestVPNTunnel::startTests()
[connection]	 Doing tunnel test 1
[server_api]	 Do ping test with timeout:  2000
[network]	 Could not resolve "checkip.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request PingTest failed:" "DnsResolveError"
[connection]	 Tunnel test  "1" failed
[server_api]	 Do ping test with timeout:  4000
[default]	 MeasurementCpuUsage stopped
[network]	 Could not resolve "checkip.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request PingTest failed:" "DnsResolveError"
[connection]	 Tunnel test  "2" failed
[server_api]	 Do ping test with timeout:  8000
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[network]	 Could not resolve "checkip.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request PingTest failed:" "DnsResolveError"
[connection]	 Tunnel test  "3" failed
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[server_api]	 "API request Session failed:" "Couldn't connect to server"
[network]	 Could not resolve "api.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request Session failed:" "DnsResolveError"
[network]	 Could not resolve "api.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request Session failed:" "DnsResolveError"
[network]	 Could not resolve "api.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request Session failed:" "DnsResolveError"
[network]	 Could not resolve "api.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request Session failed:" "DnsResolveError"
[network]	 Could not resolve "api.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request Session failed:" "DnsResolveError"
[network]	 Could not resolve "api.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request Session failed:" "DnsResolveError"
[network]	 Could not resolve "api.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request Session failed:" "DnsResolveError"
[network]	 Could not resolve "api.windscribe.com" (servers: QList() ): "Channel is being destroyed"
[server_api]	 "API request Session failed:" "DnsResolveError"
[network]	 Could not resolve "api.windscribe.com" (servers: QList() ): "Channel is being destroyed"
```

**Comment by bernerdad** [comment]
Do you have split tunneling enabled?  If yes, can you disable it and let us know if you can then connect?  Thanks!

**Comment by ltguillaume** [comment]
Ah damn, I wanted to test that before posting. Yes, I had exclusive split tunneling turned on.

Turning it off indeed resolves the connection issue.
Turning it on again immediately severs the connection.

**Comment by bernerdad** [comment]
Thanks!  Yes, we're tracking a regression in 2.7.10 that seems to occur if one has exclusive split tunnel on and no apps specified to exclude, only IPs/hostnames.

**Comment by ltguillaume** [comment]
> Thanks! Yes, we're tracking a regression in 2.7.10 that seems to occur if one has exclusive split tunnel on **and no apps specified to exclude, only IPs/hostnames**.

Yes, that is exactly the case here.

**Comment by bernerdad** [comment]
Hi.  Just a heads up that we've fixed our gaff and will release 2.7.11 asap.

**Comment by bernerdad** [comment]
Hi.  Here's an early preview of the v2.7.11 guinea pig release.  Please let us know if it resolves this issue for you.  Thanks!
https://deploy.totallyacdn.com/desktop-apps/2.7.11/Windscribe_2.7.11_guinea_pig.exe
